src/test/java/com/opentable/versionedconfig/GitServiceTest.java

```diff
@@ -172,15 +176,30 @@ public void testRevisionMetadata() throws IOException {
     public void testCredentialsMustBothBeEmptyOrNull() {
         assertThatExceptionOfType(IllegalArgumentException.class)
                 .isThrownBy(() -> VersioningService.forGitRepository(
-                        new GitProperties(null, "username", null, null, "master")))
+                        new GitProperties(Collections.emptyList(), "username", null, null, "master")))
                 .withMessageContaining("must provide username and password");
 
         assertThatExceptionOfType(IllegalArgumentException.class)
                 .isThrownBy(() -> VersioningService.forGitRepository(
-                        new GitProperties(null, null, "password", null, "master")))
+                        new GitProperties(Collections.emptyList(), null, "password", null, "master")))
                 .withMessageContaining("must provide username and password");
     }
 
+    @Test
+    public void testFallbackPull() throws Exception {
+        workFolder.create();
+        final File checkoutSpot = workFolder.newFolder("init");
+        final GitProperties gitProperties = new GitProperties(ImmutableList.of(URI.create("git://example.invalid"), remote.getLocalPath().toUri()), null, null, checkoutSpot, "master");
```

Comment:
I see this test works by checking for expected revision in the local git repo, but should we not also test that it picks the URLs in a specific order? We can't be sure by this test that `git://example.invalid` is even being considered...

Reply:
improved in latest commit